File: core/src/main/java/com/jinkyumpark/bookitout/search/apiResponse/ridi/ApiRidiBook.java
package com.jinkyumpark.bookitout.search.apiResponse.ridi;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.jinkyumpark.bookitout.search.provider.SubscriptionProvider;
import com.jinkyumpark.bookitout.search.response.searchResult.SubscriptionSearchResult;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor @AllArgsConstructor
@Getter
public class ApiRidiBook {
    private String endDatetime;
    private String author;
    private String translator;
    private String ageLimit0;
    private String title;
    private String beginDatetime;
    private String author2;
    private String isDeletedfalse;
    private String webTitle;
    @JsonProperty("b_id")
    private String id;
    private String publisher;
    private String webTitleTitle;
    private List<ApiRidiAuthor> authorsInfo;

    public SubscriptionSearchResult toSubscriptionSearchResult() {
        return SubscriptionSearchResult.builder()
                .title(this.title)
                .author(
                        this.author + (this.translator != null ? String.format(", %s (번역가)", this.translator) : "")
                )
                .cover(String.format("https://img.ridicdn.net/cover/%s/small?dpi=xhdpi", this.id))
                .link(String.format("https://select.ridibooks.com/book/%s", this.id))
                .provider(SubscriptionProvider.RIDI)
                .build();
    }
}

File: search/src/main/kotlin/com/jinkyumpark/search/subscription/apiResponse/ridi/ApiRidiBook.kt
package com.jinkyumpark.search.subscription.apiResponse.ridi

import com.fasterxml.jackson.annotation.JsonProperty
import com.jinkyumpark.search.subscription.SubscriptionProvider
import com.jinkyumpark.search.subscription.SubscriptionSearchResponse

data class ApiRidiBook(
    var endDatetime: String?,
    val author: String?,
    val translator: String?,
    val ageLimit0: String?,
    val title: String,
    val beginDatetime: String?,
    val author2: String?,
    val isDeletedfalse: String?,
    val webTitle: String?,

    @JsonProperty("b_id")
    val id: String?,

    val publisher: String?,
    val webTitleTitle: String?,
    val authorsInfo: List<ApiRidiAuthor>?,
) {
    fun toSubscriptionSearchResponse(): SubscriptionSearchResponse {
        return SubscriptionSearchResponse(
            title=title,
            author="$author, $translator (번역가)",
            cover="https://img.ridicdn.net/cover/$id/small?dpi=xhdpi",
            link="https://select.ridibooks.com/book/$id",
            provider = SubscriptionProvider.RIDI,
        )
    }
}
